components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+    showButtonContentDescription: String? = null,
+    hideButtonContentDescription: String? = null,
+    showButtonStateDescription: String? = null,
+    hideButtonStateDescription: String? = null,
+    singleLine: Boolean = true,
+    numericOnly: Boolean = false,
+    maxChars: Int? = null,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    requiredSequencesSpacing: Boolean = false,
+    textStyle: TextStyle = HmrcTheme.typography.body
+) {
+    // pattern matches a non decimal number
+    val nonDecimalPattern = remember { Regex("^([0-9]*)$") }
+
+    fun decimalPatternChecker(input: String, localValue: String) = when {
+        numericOnly && input.matches(nonDecimalPattern) -> input
+        !numericOnly -> input
+        else -> localValue
+    }
+
+    PasswordTextInputView(
+        modifier = modifier.semantics { password() },
+        value = value,
+        onInputValueChange = onInputValueChange,
+        inputFilter = { it: String, localValue: String ->
+            decimalPatternChecker(
+                it,
+                localValue
+            )
+        },
+        labelText = labelText,
+        labelContentDescription = labelContentDescription,
+        hintText = hintText,
+        hintContentDescription = hintContentDescription,
+        placeholderText = placeholderText,
+        errorText = errorText,
+        errorContentDescription = errorContentDescription,
+        showButtonContentDescription = showButtonContentDescription,
+        hideButtonContentDescription = hideButtonContentDescription,
+        showButtonStateDescription = showButtonStateDescription,
+        hideButtonStateDescription = hideButtonStateDescription,
+        maxChars = maxChars,
+        singleLine = singleLine,
+        keyboardOptions = KeyboardOptions(
+            keyboardType = if (numericOnly) KeyboardType.NumberPassword else KeyboardType.Password
+        ),
+        interactionSource = interactionSource,
+        requiredSequencesSpacing = requiredSequencesSpacing,
+        textStyle = textStyle
+    )
+}
+
+@Composable
+fun animateBorderStrokeAsState(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    focusedBorderThickness: Dp,
+    unfocusedBorderThickness: Dp
+): State<BorderStroke> {
+    val focused by interactionSource.collectIsFocusedAsState()
+    val indicatorColor = indicatorColor(enabled, isError, interactionSource)
+    val targetThickness = if (focused) focusedBorderThickness else unfocusedBorderThickness
+    val animatedThickness = if (enabled) {
+        animateDpAsState(targetThickness, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(unfocusedBorderThickness)
+    }
+    return rememberUpdatedState(
+        BorderStroke(animatedThickness.value, SolidColor(indicatorColor.value))
+    )
+}
+
+var borderStroke: State<BorderStroke>? = null
+
+@Composable
+fun CalculateBorder(
```

Comment:
I think we should make this private so that it's only able to be used within this class
```suggestion
private fun CalculateBorder(
```